src/models/user

```diff
@@ -0,0 +1,220 @@
+    },
+    providerId: {
+        type: String,
+    },
+    isVerified: {
+        type: Boolean,
+        required: true
+    },
+    createdEvents: [{
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "Event",
+
+    }],
+    followers: [{
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "User",
+    }],
+    createdFunds: [{
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "Fund"
+    }]
+
+}, { toJSON: { virtuals: true } }, { toObject: { virtuals: true } });
+
+const BaseUser = mongoose.model("User", baseUser);
+
+
+const user = mongoose.Schema({
+
+    firstName: {
+        type: string,
+        required: true
+    },
+    lastName: {
+        type: string,
+        required: true
+    },
+    // fullname: virtual property
+
+    profileImage: {
+        type: string,
+    },
+    dateOfBirth: {
+        type: string,
+        required: true
+    },
+    preferredCities: [{
+        type: string,
+        enum: cities,
+        interests: [{
+            type: string,
+            enum: listOfinterest,
+            required: true
+        }],
+        gender: {
+            type: string,
+            enum: ['male', 'female', 'other']
+        },
+        followedEvents: [{
+            type: mongoose.Schema.Types.ObjectId,
+            ref: "Event",
+
+        }],
+        followedFunds: [{
+            type: mongoose.Schema.Types.ObjectId,
+            ref: "Fund",
+        }],
+        followedUsers: [{
+            type: mongoose.Schema.Types.ObjectId,
+            ref: "User",
+        }],
+        followedOrganizations: [{
+            type: mongoose.Schema.Types.ObjectId,
+            ref: "Organization",
+
+        }]
+
+    }]
+
+})
+
+// virtual
+user.virtual('fullName').get(function () {
+    return this.firstName + ' ' + this.lastName;
+});
+
+const User = BaseUser.discriminator("User", user);
+
+
+
+
+const organization = mongoose.Schema({
+    name: {
+        type: string,
+        required: true
+    },
+    description: {
+        type: string,
+        required: true
+    },
+    coverImage: {
+        type: string,
+        required: true
+
+    },
+    categories: [{
+        type: string,
+        enum: ['No Poverty', 'Zero Hunger', 'Good Health And Well-Being',
+            'Quality Education', 'Gender Equality', 'Clean Water And Sanitation',
+            'Affordable And Clean Energy', 'Animals', 'Oceans', 'Nature',
+            'Reduced Inequalities', 'Sustainable Cities And Communities', 'Responsible Consumption And Production',
+            'Climate Action', 'Life Below Water', 'Life On Land', 'Peace', 'Youth', 'Justice']
+    }],
+    city: {
+        type: string,
+        required: true
+
+    },
+    rates: [rate],
+    // rate : virtual property
+    websiteUrl: {
+        type: string,
+    }
+
+
+});
+
+// virtual
+organization.virtual('rate').get(function () {
+    if (!this.rates.length) return 0
+    return this.rates.reduce((acc, curr) => acc + curr, 0) / rate.length
```

Comment:
Better than before but still has 2 issues: 
` return this.rates.reduce((acc, curr) => acc + curr.rate, 0) / this.rates.length`

Reply:
thank you fix it